syntax = "proto3";

package vigil;

import "google/protobuf/timestamp.proto";

import "location.proto";

// Represents a single symptom report from a user
message SymptomReport {
  string                    user_id   = 1;
  google.protobuf.Timestamp timestamp = 2;

  Location                  location  = 3; // Geographic location of the report

  map<string, int32>        symptoms  = 4; // Symptoms with per-symptom severity (1-3)

  string                    notes     = 5; // Optional free-text notes

  string                    cause     = 6; // Inferred most likely cause.
}
